base/deprecated.jl

```diff
@@ -1417,6 +1417,20 @@ function LibGit2.set_remote_url(path::AbstractString, url::AbstractString; remot
     LibGit2.set_remote_url(path, remote, url)
 end
 
+function replace(s::AbstractString, pat, f, n::Integer)
```

Comment:
Would be good to add a comment explaining that `replace_new` needs to be renamed to `replace` when removing this deprecation (and make sure tests will fail if that's not the case).